util/hash.cc

```diff
@@ -80,4 +80,19 @@ uint64_t Hash64(const char* data, size_t n) {
   return XXH3p_64bits(data, n);
 }
 
+uint64_t GetSlicePartsNPHash64(const SliceParts& data) {
+  // TODO(ajkr): use XXH3 streaming APIs to avoid the copy/allocation.
```

Comment:
XXH3(p) is highly tuned for variable length known at the start of hashing or compile-time length, because its behavior varies substantially depending on length. Therefore its streaming support is complex and might not be very fast on small-to-medium data. It's funny here that we know the overall length up front; we just have a split buffer.

I would expect something with more uniform structure, like recent generations of Murmur hashes, to work better for split / streaming hashes of small-to-medium data size.

Reply:
It needs more investigation/benchmarking and possibly a new dependency, so I'll probably defer looking at other hash functions.